File: member-service/src/main/java/com/hedvig/integration/notificationService/NotificationService.java
package com.hedvig.integration.notificationService;

import com.hedvig.integration.notificationService.dto.CancellationEmailSentToInsurerRequest;
import com.hedvig.integration.notificationService.dto.InsuranceActivationDateUpdatedRequest;
import org.jetbrains.annotations.NotNull;

import java.util.Map;

public interface NotificationService {

  void cancellationEmailSentToInsurer(Long memberId, CancellationEmailSentToInsurerRequest body);

  void insuranceActivated(Long memberId);

  void insuranceActivationDateUpdated(Long memberId, InsuranceActivationDateUpdatedRequest body);

  void insuranceReminder(int NumberOfDaysFromToday);

  void deleteCustomer(@NotNull String memberId);

  void updateCustomer(@NotNull String memberId, @NotNull Map<String, Object> traitsMap);
}

File: member-service/src/main/java/com/hedvig/integration/notificationService/NotificationService.kt
package com.hedvig.integration.notificationService

import com.hedvig.integration.notificationService.dto.CancellationEmailSentToInsurerRequest
import com.hedvig.integration.notificationService.dto.InsuranceActivationDateUpdatedRequest

interface NotificationService {
    fun cancellationEmailSentToInsurer(memberId: Long, body: CancellationEmailSentToInsurerRequest)
    fun insuranceActivated(memberId: Long)
    fun insuranceActivationDateUpdated(memberId: Long, body: InsuranceActivationDateUpdatedRequest)
    fun insuranceReminder(NumberOfDaysFromToday: Int)
    fun deleteCustomer(memberId: String)
    fun updateCustomer(memberId: String, traitsMap: Map<String, Any?>)
    fun updatePhoneNumber(eventId: String, memberId: String, phoneNumber: String)
}
